src/parallelization/hpxsimulator.cpp

```diff
@@ -6,12 +6,16 @@
 namespace LibGeoDecomp {
 namespace HpxSimulatorHelpers {
 
+hpx::lcos::local::promise<void> initPromise;
 std::map<std::string, hpx::lcos::local::promise<std::vector<double> > > localUpdateGroupSpeeds;
 std::map<std::string, hpx::lcos::local::promise<std::vector<double> > > myGlobalUpdateGroupSpeeds;
 std::map<std::string, hpx::lcos::local::promise<std::vector<std::size_t> > > myLocalityIndices;
 
 hpx::future<std::vector<double> > getUpdateGroupSpeeds(const std::string& basename)
```

Comment:
On 02/12/2016 02:11 PM, Andreas Schäfer wrote:
> In src/parallelization/hpxsimulator.cpp
> <https://github.com/STEllAR-GROUP/libgeodecomp/pull/3#discussion_r52738929>:
> 
>> @@ -6,12 +6,16 @@
>>  namespace LibGeoDecomp {
>>  namespace HpxSimulatorHelpers {
>>  
>> +hpx::lcos::local::promise<void> initPromise;
>>  std::map<std::string, hpx::lcos::local::promise<std::vector<double> > > localUpdateGroupSpeeds;
>>  std::map<std::string, hpx::lcos::local::promise<std::vector<double> > > myGlobalUpdateGroupSpeeds;
>>  std::map<std::string, hpx::lcos::local::promise<std::vector<std::size_t> > > myLocalityIndices;
>>  
>>  hpx::future<std::vector<double> > getUpdateGroupSpeeds(const std::string& basename)
> 
> I know I should have asked this in the previous PR, but apparently did
> overlook this. What's the rationale here behind returning a future
> instead of the actual value?

You safe an explicit .get on the future, which should be avoided. by
returning the future, the result will be sent across the wire (or not)
whenever the result is ready.

> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/STEllAR-GROUP/libgeodecomp/pull/3/files#r52738929>.
> 


-- 
Thomas Heller
Friedrich-Alexander-Universität Erlangen-Nürnberg
Department Informatik - Lehrstuhl Rechnerarchitektur
Martensstr. 3
91058 Erlangen
Tel.: 09131/85-27018
Fax:  09131/85-27912
Email: thomas.heller@cs.fau.de